pkg/queue/protobuf_stats_reporter.go

```diff
@@ -33,55 +33,55 @@ const (
 
 // ProtobufStatsReporter structure represents a protobuf stats reporter.
 type ProtobufStatsReporter struct {
-	reportingPeriod time.Duration
-	startTime       time.Time
-	stat            atomic.Value
-	podName         string
+	startTime time.Time
+	stat      atomic.Value
+	podName   string
+
+	// RequestCount and ProxiedRequestCount need to be divided by the reporting period
+	// they were collected over to get a "per-second" value.
+	reportingPeriodSeconds float64
 }
 
 // NewProtobufStatsReporter creates a reporter that collects and reports queue metrics.
 func NewProtobufStatsReporter(pod string, reportingPeriod time.Duration) *ProtobufStatsReporter {
 	return &ProtobufStatsReporter{
-		reportingPeriod: reportingPeriod,
-		startTime:       time.Now(),
-		stat:            atomic.Value{},
-		podName:         pod,
+		startTime: time.Now(),
+		podName:   pod,
+
+		reportingPeriodSeconds: reportingPeriod.Seconds(),
 	}
 }
 
 // Report captures request metrics.
 func (r *ProtobufStatsReporter) Report(stats network.RequestStatsReport) {
-	// Requests per second is a rate over time while concurrency is not.
-	rp := r.reportingPeriod.Seconds()
 	r.stat.Store(metrics.Stat{
-		PodName:                          r.podName,
-		RequestCount:                     stats.RequestCount / rp,
-		ProxiedRequestCount:              stats.ProxiedRequestCount / rp,
+		PodName:       r.podName,
+		ProcessUptime: time.Since(r.startTime).Seconds(),
+
+		// RequestCount and ProxiedRequestCount are a rate over time while concurrency is not.
+		RequestCount:                     stats.RequestCount / r.reportingPeriodSeconds,
+		ProxiedRequestCount:              stats.ProxiedRequestCount / r.reportingPeriodSeconds,
 		AverageConcurrentRequests:        stats.AverageConcurrency,
 		AverageProxiedConcurrentRequests: stats.AverageProxiedConcurrency,
-		ProcessUptime:                    time.Since(r.startTime).Seconds(),
 	})
 }
 
-// Handler returns an uninstrumented http.Handler used to serve stats registered by this
-// ProtobufStatsReporter.
-func (r *ProtobufStatsReporter) Handler() http.Handler {
```

Comment:
FWIW: I had this on purpose seeing as the concurrency_reporter does more than just being a HTTP handler. I thought the separation is neater if we generate a handler specifically. Certainly a nit though.

Reply:
Ehr, nevermind, these are completely different files :joy:. Maybe the same reasoning applies though.

Reply:
I have to admit, I can't really see the advantage of `Handler() http.Handler` vs. `ServeHTTP()` in terms of encapsulation. In both cases you have a method in the interface to return a http handler/handlerfuc, it's just one happens to actually implement http.Handler and therefore can be directly used. Also, 100% of what this class does is report metrics via protobuf over HTTP so.. :)